src/bits/shared.h

```diff
@@ -53,6 +53,19 @@ namespace detail {
     template <typename FN, typename... ARGS>
     using invoke_result_t = typename invoke_result<FN, ARGS...>::type;
 
+    // Test if two variables is at the same address.
+    template <typename T, typename S>
+    bool is_at_same_address(T, S)
+    {
+        return false;
+    }
+
+    template <typename T>
+    bool is_at_same_address(const T &x, const T &y)
+    {
+        return &x == &y;
```

Comment:
And use std::addressof in C++17.

Reply:
Unfortunately I've restricted myself to C++14. Besides it would still complain about comparing different types.
Hope the new version works for you